packages/flutter/test/cupertino/text_field_test.dart

```diff
@@ -250,6 +250,39 @@ void main() {
     },
   );
 
+  testWidgets('Look Up shows up on iOS only', (WidgetTester tester) async {
+      final TextEditingController controller = TextEditingController(
+        text: 'Test',
+      );
+      await tester.pumpWidget(
+        CupertinoApp(
+          home: Center(
+            child: CupertinoTextField(
+              controller: controller,
+            ),
+          ),
+        ),
+      );
+
+      final bool isTargetPlatformiOS = defaultTargetPlatform == TargetPlatform.iOS;
+
+      // Long press to put the cursor after the "s".
+      const int index = 3;
+      await tester.longPressAt(textOffsetToPosition(tester, index));
+      await tester.pump();
+
+      // Double tap on the same location to select the word around the cursor.
+      await tester.tapAt(textOffsetToPosition(tester, index));
+      await tester.pump(const Duration(milliseconds: 50));
+      await tester.tapAt(textOffsetToPosition(tester, index));
+      await tester.pumpAndSettle();
+
+      expect(controller.selection, const TextSelection(baseOffset: 0, extentOffset: 4));
+      expect(find.text('Look Up'), isTargetPlatformiOS? findsOneWidget : findsNothing);
```

Comment:
Maybe also tap the button and test that it calls your platform channel method with the current selection?  You can listen to it like this more or less:

```dart
    String? lastLookUp;
    setUp(() {
      TestDefaultBinaryMessengerBinding.instance.defaultBinaryMessenger
        .setMockMethodCallHandler(SystemChannels.platform, (MethodCall methodCall) async {
          if (methodCall.method == 'LookUp.invoke') {
            expect(methodCall.arguments, isA<String>());
            lastLookUp = methodCall.arguments as String;
          }
        });
    });

    tearDown(() {
      TestDefaultBinaryMessengerBinding.instance.defaultBinaryMessenger
          .setMockMethodCallHandler(SystemChannels.platform, null);
    });
```